bundler/lib/bundler/endpoint_specification.rb

```diff
@@ -129,13 +128,6 @@ def parse_metadata(data)
 
     def build_dependency(name, requirements)
       Gem::Dependency.new(name, requirements)
-    rescue ArgumentError => e
-      raise unless e.message.include?(ILLFORMED_MESSAGE)
-      puts # we shouldn't print the error message on the "fetching info" status line
-      raise GemspecError,
-        "Unfortunately, the gem #{name} (#{version}) has an invalid " \
-        "gemspec.\nPlease ask the gem author to yank the bad version to fix " \
-        "this issue. For more information, see http://bit.ly/syck-defaultkey."
```

Comment:
Another observation about this is that although we seem to have been warning this situation for a long time, as far as I can see, this code is only reached from parsing the compact index "info" endpoint, which as far as I understand, can't serve this "weird dependencies".

Only situation where I think we could still be bitten by this is when unpacking a gem package and reading a corrupted gemspec inside it. It'd be nice to find such a corrupted gem package and run some experiments on it.